mrmustard/lab_dev/circuit_components.py

```diff
@@ -471,12 +471,16 @@ def to_fock(self, shape: int | Sequence[int] | None = None) -> CircuitComponent:
                 defaults to the value of ``AUTOSHAPE_MAX`` in the settings.
         """
         fock = Fock(self.fock(shape, batched=True), batched=True)
-        fock._original_bargmann_data = self.representation.data
+        try:
+            fock._original_bargmann_data = self.representation.triple
+        except AttributeError:
+            fock._original_bargmann_data = None
         try:
             ret = self._getitem_builtin(self.modes)
             ret._representation = fock
         except TypeError:
             ret = self._from_attributes(fock, self.wires, self.name)
+        ret.manual_shape = ret.representation.array.shape[1:]
```

Comment:
this should be handled by the cached_property right?

Reply:
yes you're right, what's the right way to do it?

Reply:
I believe you just delete it (so that the next time it's called, it gets re-computed):
```suggestion
        del ret.manual_shape
```

from [the docs](https://docs.python.org/3/library/functools.html#functools.cached_property)
> The cached value can be cleared by deleting the attribute. This allows the cached_property method to run again.

Reply:
hmm, but I want to update it, not delete it. Maybe I should do it entry-wise?

Reply:
the next time it's invoked, it'll [re-evaluate](https://github.com/XanaduAI/MrMustard/blob/4ac64c1be84ee98ba7290755e52c8c3e5f11a0db/mrmustard/lab_dev/circuit_components.py#L202) (update) it, and it should be correct. Example:
```python
from functools import cached_property

class X:
    def __init__(self, x):
        self.x = x

    @cached_property
    def xsum(self):
        return sum(self.x)
```

```pycon
>>> foo = X([1,2,3])
>>> foo.xsum
6
>>> foo.x = [4,5,6]
>>> foo.xsum
6  # BAD! didn't update
>>> del foo.xsum
>>> foo.xsum
15 
```

Reply:
oooh silly me. I had forgotten it already takes the shape of the array if it exists.